miniapps/nurbs/nurbs_naca_cmesh.cpp

```diff
@@ -0,0 +1,798 @@
+      patch4(1,1,1) = FlairBoundDist(flair, boundary_dist, patch4(1,1,0));;
+
+      // Refine
+      patch4.DegreeElevate(0, order-1);
+      patch4.KnotInsert(0, *kv4);
+      patch4.DegreeElevate(1, order-1);
+      patch4.KnotInsert(1, *kvr);
+   }
+
+   // Apply angle of attack
+   patch0.Rotate2D(-aoa);
+   patch1.Rotate2D(-aoa);
+   patch2.Rotate2D(-aoa);
+   patch3.Rotate2D(-aoa);
+   patch4.Rotate2D(-aoa);
+
+   // 5. Print mesh to file
+
+   // Open mesh output file
+   string mesh_file;
+   mesh_file.append(msh_path);
+   mesh_file.append(msh_filename);
+   mesh_file.append(".mesh");
+   ofstream output(mesh_file.c_str());
+
+   // File header
+   output<<"MFEM NURBS mesh v1.0"<<endl;
+   output<< endl << "# " << mdim << "D C-mesh around a symmetric NACA foil section"
+         << endl << endl;
+   output<< "dimension"<<endl;
+   output<< mdim <<endl;
+   output<< endl;
+
+   // NURBS patches defined as elements
+   output << "elements"<<endl;
+   output << "5"<<endl;
+   output << "1 3 0 1 5 4" << endl;   // Lower wake
+   output << "1 3 1 2 6 5" << endl;   // Lower tail
+   output << "1 3 2 3 7 6" << endl;   // Tip
+   output << "1 3 3 1 8 7" << endl;   // Upper tail
+   output << "1 3 1 0 9 8" << endl;   // Upper wake
+   output << endl;
+
+   // Boundaries
+   output << "boundary" <<endl;
+   output << "10" <<endl;
+   output << "1 1 5 4" << endl;   // Bottom
+   output << "1 1 6 5" << endl;   // Bottom
+   output << "2 1 7 6" << endl;   // Inflow
+   output << "3 1 8 7" << endl;   // Top
+   output << "3 1 9 8" << endl;   // Top
+   output << "4 1 4 0" << endl;   // Outflow
+   output << "4 1 0 9" << endl;   // Outflow
+   output << "5 1 1 2" << endl;   // Foil section
+   output << "5 1 2 3" << endl;   // Foil section
+   output << "5 1 3 1" << endl;   // Foil section
+   output << endl;
+
+   // Edges
+   output <<"edges"<<endl;
+   output <<"15"<<endl;
+   output << "0 0 1"<<endl;
+   output << "1 1 2"<<endl;
+   output << "2 2 3"<<endl;
+   output << "3 3 1"<<endl;
+
+   output << "0 4 5"<<endl;
+   output << "1 5 6"<<endl;
+   output << "2 6 7"<<endl;
+   output << "3 7 8"<<endl;
+   output << "0 9 8"<<endl;
+
+   output << "4 0 4"<<endl;
+   output << "4 1 5"<<endl;
+   output << "4 2 6"<<endl;
+   output << "4 3 7"<<endl;
+   output << "4 1 8"<<endl;
+   output << "4 0 9"<<endl;
+   output << endl;
+
+   // Vertices
+   output << "vertices" << endl;
+   output << 10 << endl;
+   output << endl;
+
+   // Patches
+   output<<"patches"<<endl;
+   output<<endl;
+
+   output << "# Patch 0 " << endl;
+   patch0.Print(output); output<<endl;
+   output << "# Patch 1 " << endl;
+   patch1.Print(output); output<<endl;
+   output << "# Patch 2 " << endl;
+   patch2.Print(output); output<<endl;
+   output << "# Patch 3 " << endl;
+   patch3.Print(output); output<<endl;
+   output << "# Patch 4 " << endl;
+   patch4.Print(output); output<<endl;
+
+   // Close
+   output.close();
+   delete kv0;
```

Comment:
This is a lot of `delete` calls. A simpler implementation might be to use `unique_ptr`. For example, see miniapps/solvers/div_free_solver.hpp.

Reply:
Interesting suggestion! I have looked into it. I think that `std::unique_ptr` is not compatible with some of the constructors of `NURBSPatch` such as 

```
NURBSPatch(const KnotVector *kv0, const KnotVector *kv1, int dim_);
```

Therefor I think we will get a mix of 'normal' pointers and `std::unique_ptr` which I think is undesirable. I will leave it in its current state.

Reply:
I am fine with the current state and do not think it is necessary to change to `unique_ptr` for this PR. A normal, raw pointer can be obtained by calling `.get()`: https://en.cppreference.com/w/cpp/memory/unique_ptr/get

Reply:
I think moving to `unique_ptr` will be better. @JacobLotz, can you see if you can make it work with `unique_ptr::get` as Dylan suggested?

Reply:
@dylan-copeland @pazner with `.get`, the implementation actually was quite easy. Thanks for this suggestion!